modules/renderer/features.js

```diff
@@ -479,17 +483,17 @@ export function rendererFeatures(context) {
 
 
     features.isHiddenFeature = function(entity, resolver, geometry) {
-        if (!_hidden.length) return false;
```

Comment:
It is a performance optimization: if all Map Features checkboxes are checked (`_hidden` = unchecked boxes = disabled pass rules), then this feature is not hidden. But since Date Range is a different filter outside of the Map Features checkboxes, that is not accurate.

I think you're mistaken about `hidden()` It accepts a rule key (e.g. *service_roads*) and returns whether those features' rule is to be skipped. The tests for each feature are `features.isHiddenFeature` `features.isHiddenChild` `features.hasHiddenConnections` and `features.isHidden`

But how about this:
- I have reinstated the `if (!_hidden.length) return false;` conditions, which would make the rules processing faster when all boxes are checked
- then put `featureFitsDateRange()` before that test, in each of these four places

Reply:
Ah, you’re right. In the future, if we allow clearing the start and end date fields without a fallback value (as with the photo overlay date filter), then `featureFitsDateRange()` could achieve a similar optimization by bailing if the dates are unset.

Reply:
You were right about removing the `_hidden.length` checks. With them in place, vertices of hidden features would become unhidden if all the map features were enabled. I missed that case because I have Boundaries disabled by default.